src/routes/mail.ts

```diff
@@ -1,35 +1,31 @@
-import express from 'express';
+import express, { Router } from 'express';
 import MailService from '@/services/MailService';
 import Mail from '@/interfaces/Mail';
-import logger from '@/helper/logger';
 
 const router = express.Router();
-const mailService = new MailService('MailService');
 
-router.post('/', (req, res) => {
-	const mail: Mail = {
-		to: req.body.to,
-		replyTo: req.body.replyTo,
-		subject: req.body.subject,
-		text: req.body.text,
-		html: req.body.html,
-		attachments: req.body.attachments,
-	};
+export default (mailService: MailService): Router => {
+	router.post('/', async (req, res, next) => {
+		try {
+			const mail: Mail = {
+				to: req.body.to,
+				replyTo: req.body.replyTo,
+				subject: req.body.subject,
+				text: req.body.text,
+				html: req.body.html,
+				attachments: req.body.attachments,
+			};
 
-	if (req.body.from) {
-		mail.from = req.body.from;
-	}
+			if (req.body.from) {
+				mail.from = req.body.from;
+			}
 
-	mailService
-		.send(req.body.platformId, mail, req.body.to)
-		.then((response: any) => {
-			res.send('Mail queued.');
-			logger.info(response);
-		})
-		.catch((error: Error) => {
-			logger.error({ error, platformId: req.body.platformId, mail, to: req.body.to });
-			res.status(500).send('Request failed.');
-		});
-});
+			await mailService.send(req.body.platformId, mail, req.body.to);
```

Comment:
does the mail service require separate receiver param behind mail?

Reply:
I think this is the 3rd param `req.body.to`.